tests_recording/test_api.py

```diff
@@ -20,42 +20,107 @@
+    record_requests_for_all_methods,
+    replace_module_match,
+)
+from requre.helpers.tempfile import TempFile
+from requre.helpers.simple_object import Simple
+from requre.helpers.files import StoreFiles
+from requre.helpers.git.pushinfo import PushInfoStorageList
+from requre.helpers.git.fetchinfo import FetchInfoStorageList
+from requre.helpers.git.repo import Repo
 
 from packit.api import PackitAPI
-from tests_recording.testbase import PackitUnittestOgr
+from tests_recording.testbase import PackitTest
 
 
-@unittest.skip("Not working yet")
-class ProposeUpdate(PackitUnittestOgr):
-    def setUp(self):
-        if (
-            hasattr(rebasehelper, "VERSION")
-            and int(rebasehelper.VERSION.split(".")[1]) >= 19
-        ):
-            self.cassette.data_miner.key = "rebase-helper>=0.19"
-        else:
-            self.cassette.data_miner.key = "rebase-helper<0.19"
-        self.cassette.data_miner.data_type = DataTypes.Dict
+@record_requests_for_all_methods()
+@apply_decorator_to_all_methods(
+    replace_module_match(
+        what="packit.utils.run_command_remote", decorate=Simple.decorator_plain()
+    )
+)
+@apply_decorator_to_all_methods(
+    replace_module_match(
+        what="packit.fedpkg.FedPKG.clone",
+        decorate=StoreFiles.where_arg_references(
+            key_position_params_dict={"target_path": 2}
+        ),
+    )
+)
+@apply_decorator_to_all_methods(
+    replace_module_match(
+        what="git.repo.base.Repo.clone_from",
+        decorate=StoreFiles.where_arg_references(
+            key_position_params_dict={"to_path": 2},
+            return_decorator=Repo.decorator_plain,
+        ),
+    )
+)
+@apply_decorator_to_all_methods(
+    replace_module_match(
+        what="git.remote.Remote.push", decorate=PushInfoStorageList.decorator_plain()
+    )
+)
+@apply_decorator_to_all_methods(
+    replace_module_match(
+        what="git.remote.Remote.fetch", decorate=FetchInfoStorageList.decorator_plain()
+    )
+)
+@apply_decorator_to_all_methods(
+    replace_module_match(what="tempfile.mkdtemp", decorate=TempFile.mkdtemp())
+)
+@apply_decorator_to_all_methods(
+    replace_module_match(what="tempfile.mktemp", decorate=TempFile.mktemp())
+)
+# Be aware that decorator stores login and token to test_data, replace it by some value.
+# Precommit hook should do it for you
+@apply_decorator_to_all_methods(
+    replace_module_match(
+        what="copr.v3.helpers.config_from_file", decorate=Simple.decorator_plain()
+    )
+)
+class ProposeUpdate(PackitTest):
+    def cassette_setup(self, cassette):
+        # cassette.data_miner.key = (
+        #    f'github-{pkg_resources.get_distribution("PyGithub").version}'
+        # )
+        # if (
+        #    hasattr(rebasehelper, "VERSION")
+        #    and int(rebasehelper.VERSION.split(".")[1]) >= 19
+        # ):
+        #    cassette.data_miner.key = "rebase-helper>=0.19"
+        # else:
+        #    cassette.data_miner.key = "rebase-helper<0.19"
```

Comment:
do we need this code?

Reply:
Yep, could be removed, or let there as an example in a case this already happen. what do you think, should I remove it, or let it there?

Reply:
without comments, it's hard for me to understand what the code does actually, so if you want to keep it, please provide comments for the use case

Reply:
the code were removed and added instruction to readme file